### PR TITLE
Add Theory of Operations Documentation for Image module

### PR DESCRIPTION
**Pull request by joschock**
## Description

Add Theory of Operations Documentation for Image module.

- [ ] Impacts functionality?
- [ ] Impacts security?
- [ ] Breaking change?
- [ ] Includes tests?
- [x] Includes documentation?

## How This Was Tested

Mdbook properly renders.

## Integration Instructions

N/A

### CODE REVIEW

File: docs/src/SUMMARY.md [hunk 1]
```diff
@@ -37,7 +37,8 @@
   - [Protocol Database](dxe_core/protocol_database.md)
   - [UEFI Driver Model](dxe_core/driver_model.md)
   - [Memory Management]()
-  - [Image Loading and Dispatcher](dxe_core/image_and_dispatcher.md)
+  - [Image Loading and Execution](dxe_core/images.md)
+  - [Dispatcher](dxe_core/dispatcher.md)
```
Comment (Javagedes) [inline docs/src/SUMMARY.md:41]: nit: You could leave this empty similar to how we are doing the other un-documented sections. I'm not worried about it though, if you want to keep it as is.

File: cspell.yml [hunk 2]
```diff
@@ -47,5 +47,6 @@ words:
   - "structs"
   - "uncacheable"
   - "wakeup"
+  - "yeilder"
```
Comment (os-d) [inline cspell.yml]: Should be `yielder`, right? May not fail spell check then :)

File: docs/src/dxe_core/images.md [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+to "exit" an image, which would otherwise require architecture-specific processor context save/and restore operations
+(i.e. "SetJump" and "LongJump"). In addition, the coroutine approach allows for better isolation of the image stack from
+the Rust DXE Core stack.
+
+When `core_start_image` is called, it first allocates a stack for the new image context, and then constructs a
+corosensei [`Coroutine`](https://docs.rs/corosensei/latest/corosensei/struct.Coroutine.html) structure. The coroutine is
+initialized with a closure that is used to invoke the image entry point.
+
+Once the `Coroutine` has been constructed, `core_start_image` passes control to it via the [`resume()`](https://docs.rs/corosensei/latest/corosensei/struct.Coroutine.html#method.resume)
+method. Control then passes to the coroutine closure, which stores the coroutine [`Yeilder`](https://docs.rs/corosensei/latest/corosensei/struct.Yielder.html)
```
Comment (os-d) [inline docs/src/dxe_core/images.md]:
```suggestion
method. Control then passes to the coroutine closure, which stores the coroutine [`Yielder`](https://docs.rs/corosensei/latest/corosensei/struct.Yielder.html)
```

File: docs/src/dxe_core/images.md [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+that serves as the context for the coroutine and then calls the `entry_point` of the image. Once the image entry point
+completes and returns, the coroutine calls the `exit()` method, which uses the coroutine yielder to pass control back to
+the point in `core_start_image` immediately following the point where `resume()` was invoked to start the co-routine.
+
+Note that this process can be recursive - for example, `core_start_image` can be invoked to start execution of UEFI
+Shell, which may in turn call `core_start_image` to invoke an application. Storing the `Yeilder` context for each image
```
Comment (os-d) [inline docs/src/dxe_core/images.md]:
```suggestion
Shell, which may in turn call `core_start_image` to invoke an application. Storing the `Yielder` context for each image
```

File: cspell.yml (hunk 2 above)
Comment (joschock) [reply cspell.yml]: hmm. I thought I cut and pasted this from corosensei; but turns out I was just incautious. Will fix.

File: docs/src/SUMMARY.md (hunk 1 above)
Comment (joschock) [reply docs/src/SUMMARY.md:41]: I'm planning on doing the dispatcher next, so it's a pretty transient state. Will leave it for now unless it's really bothersome.

File: cspell.yml (hunk 2 above)
Comment (joschock) [reply cspell.yml]: fixed in latest.

File: docs/src/dxe_core/images.md (hunk 3 above)
Comment (joschock) [reply docs/src/dxe_core/images.md]: fixed in latest

File: docs/src/dxe_core/images.md (hunk 4 above)
Comment (joschock) [reply docs/src/dxe_core/images.md]: fixed in latest

**Pull request merged**